presto-jmx/src/test/java/com/facebook/presto/connector/jmx/TestJmxSplitManager.java

```diff
@@ -118,6 +132,59 @@ public void testRecordSetProvider()
         }
     }
 
+    @Test
+    public void testHistoryRecordSetProvider()
+            throws Exception
+    {
+        for (SchemaTableName schemaTableName : metadata.listTables(SESSION, HISTORY_SCHEMA_NAME)) {
+            // wait for at least two samples
+            List<Long> timeStamps = ImmutableList.of();
+            for (int waited = 0; waited < TIMEOUT_TIME; waited += SLEEP_TIME) {
+                RecordSet recordSet = getRecordSet(schemaTableName);
+                timeStamps = readTimeStampsFrom(recordSet);
+                if (timeStamps.size() >= 2) {
+                    break;
+                }
+                Thread.sleep(SLEEP_TIME);
```

Comment:
this is not a unit test, but integration test of whole history dumping. I don't like this either, but we have to wait for some action to happen. Do you have other idea how to test this?